runtime/bin/socket_base_posix.cc

```diff
@@ -151,13 +151,27 @@ intptr_t SocketBase::ReceiveMessage(intptr_t fd,
   size_t num_messages = 0;
   while (cmsg != nullptr) {
     num_messages++;
+#ifndef __GLIBC__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wsign-compare"
+#endif
     cmsg = CMSG_NXTHDR(&msg, cmsg);
```

Comment:
Consider introducing a macro that wraps `CMSG_NXTHDR`, e.g. 

```cpp
// We wrap CMSG_NXTHDR to suppress sign-compare warnings which occur on musl.
#define NEXT_HEADER(mhdr, cmsg)                     \ 
  _Pragma("clang diagnostic push")                   \
  _Pragma("clang diagnostic ignored -Wsign-compare") \
  CMSG_NXTHDR(mhdr, cmsg)                            \
  _Pragma("clang diagnostic pop")
```

Instead of suppressing the error around each call site.

Reply:
@mraleph Good point. Updated PR.